toplevel/vernac.ml

```diff
@@ -128,7 +128,8 @@ let rec interp_vernac sid po (loc,com) =
 
         (* We could use a more refined criteria depending on the
            vernac. For now we imitate the old approach. *)
-        let print_goals = not (!Flags.batch_mode || is_query v) in
+        let print_goals = not (!Flags.batch_mode || is_query v ||
+                               not (Proof_global.there_are_pending_proofs ())) in
```

Comment:
This is hard to read. Why not rather:

    let print_goals = not (!Flags.batch_mode || is_query v) && Proof_global.there_are_pending_proofs () in

?